sailthru_content/sync_library.py

```diff
@@ -0,0 +1,224 @@
++--------------------------------+-------------------------------------------------------+
+| --lms_url                      | Url of LMS (default http://courses.edx.org            |
++--------------------------------+-------------------------------------------------------+
+| --email_report                 | Email address to sent batch report to                 |
++--------------------------------+-------------------------------------------------------+
+| --type                         | A choice of "programs", "courses" or "all"            |
++--------------------------------+-------------------------------------------------------+
+
+To get access to the Course Discovery API, you need either an existing access token, or you can specify the
+oauth_host, oauth_key and oauth_secret.
+
+Sample command to load course content in batch:
+
+sync_library.py --oauth_key=<api key> --oauth_secret=<api secret>
+   --oauth_host=https://api.edx.org/oauth2/v1
+   --sailthru_key=<sailthru key> --sailthru_secret=<sailthru secret>
+   --content_api_url=https://prod-edx-discovery.edx.org/api/v1/
+   --lms_url=https://courses.edx.org
+   --fixups=fixups.csv
+   --type=courses
+   --email_report=somebody@edx.org
+   upload
+
+"""
+
+
+def setup_logging():
+    """ Setup logging.
+    """
+    logging.basicConfig(format='%(asctime)s - %(levelname)s - %(message)s', level=logging.DEBUG)
+
+
+def load_fixups(filename):
+    """
+    Read list of fixups.
+
+    The fixups file should be a three column csv with any fields that should be overriden
+
+    Each line should have:
+        course_run,field_name,value
+    """
+
+    if not filename:
+        return []
+
+    with open(filename, 'rb') as f:
+        reader = csv.reader(f)
+        return list(reader)
+
+
+def main(argv):
+
+    args = get_args(argv)
+
+    sailthru_service = SailthruApiService(args.sailthru_key, args.sailthru_secret, args.lms_url)
+
+    if args.command == 'list':
+        sailthru_service.list()
+    elif args.command in ('upload', 'preview'):
+        fixups_collection = None
+        if args.fixups:
+            fixups_collection = load_fixups(args.fixups)
+
+        catalog_api_service = CatalogApiService(
+            args.access_token,
+            args.oauth_host,
+            args.oauth_key,
+            args.oauth_secret,
+            args.content_api_url
+        )
+
+        translation_service = SailthruTranslationService(catalog_api_service, args.lms_url, fixups_collection)
+
+        sailthru_items = []
+        if args.type in ('courses', 'all'):
+            sailthru_items.extend(translation_service.translate_courses())
+
+        if args.type in ('programs', 'all'):
+            sailthru_items.extend(translation_service.translate_programs())
+
+        if args.command == 'upload':
+            sailthru_service.upload(sailthru_items, args.report_email)
+        else:
+            print (json.dumps(sailthru_items))
```

Comment:
@schenedx a note (maybe in the README) about which version of the Python the script supports would be good to have.